common/src/test.rs

```diff
@@ -17,9 +19,19 @@ pub use openmls::*;
 
 static INIT: OnceLock<()> = OnceLock::new();
 
+use toxiproxy_rust::TOXIPROXY;
+
+static TOXIPROXY_TEST_LOCK: LazyLock<sync::Mutex<()>> = LazyLock::new(|| sync::Mutex::new(()));
+
+// TODO: can add this to the macro
+pub async fn toxiproxy_test<T, F: AsyncFn() -> T>(f: F) -> T {
```

Comment:
test failures were happening b/c of race conditions with toxiproxy. multiple toxiproxy test would configure the global toxiproxy settings, causing test to flake randomly

 changing all toxiproxy tests to go through this closure should increase the reliability of these tests (only one toxi test at a time)